pkg/dockerclient/shell.go

```diff
@@ -227,6 +228,43 @@ func (c *ShellClient) GetContainerLogs(containerID string) (string, error) {
 	return runResult.Output, err
 }
 
+// AwaitContainerHealth blocks until the given container is healthy or the timeout passes
+func (c *ShellClient) AwaitContainerHealth(containerID string, timeout time.Duration) error {
+	containerHealthy := make(chan error, 1)
+
+	go func() {
+		for {
+
+			// inspect the container's health, return if it's healthy
+			runResult, err := c.runCommand(nil, "docker inspect --format '{{json .State.Health.Status}}' %s", containerID)
+			if err == nil {
+				stdoutLines := strings.Split(runResult.Output, "\n")
+				lastStdoutLine := c.getLastNonEmptyLine(stdoutLines)
+
+				if lastStdoutLine == "\"healthy\"" {
+					containerHealthy <- nil
+					return
+				}
+			}
+
+			// wait a bit before retrying
+			c.logger.DebugWith("Container not healthy yet, retrying soon", "inspectOutput", runResult.Output)
+			time.Sleep(time.Second)
```

Comment:
I would go the extra mile here and make this timeout, starting with say 100ms (then 200, then 400) as this affects the snappiness of deploy